classes/Address.php

```diff
@@ -109,6 +109,9 @@ class AddressCore extends ObjectModel
     /**
      * @see ObjectModel::$definition
      */
+    
+    //when you override this class, do not create a field with allow_null=>true 
+    //becuase it will give you exception on checkout address step
```

Comment:
```diff
- becuase
+ because
```

By all means, we strongly discourage people to use overrides, and this feature will probably be removed in the next major version as this make the upgrade of shops quite impossible.

Reply:
if we remove overrides then how will we deal with highly customized themes? for an example; i need to show discounts as percent, even its not a percent discount, it should be shown as percent so whats the best approach to this situation? maybe i can override frontController and inject my own classes and pass to templates etc etc. instead of removing this feature, core classes should be more overridable  otherwise we will kill bunch of useful module, i mean there is no filter in search and manufacture page but there is a module for this situation and probably it have to override 1 or 2 controller and is there any better way to do this thing and or am i just am i blind af to see how things goes?

Reply:
the only recommended way to do it is to use hooks.